packages/hoppscotch-common/src/components/collections/ImportExport.vue

```diff
@@ -1,361 +1,475 @@
+    icon: IconGithub,
+    disabled: !currentUser.value
+      ? true
+      : currentUser.value.provider !== "github.com",
+    title: t("export.create_secret_gist"),
+    applicableTo: ["personal-workspace", "team-workspace"],
+    isLoading: isHoppGistCollectionExporterInProgress,
+  },
+  action: async () => {
+    isHoppGistCollectionExporterInProgress.value = true
+
+    const collectionJSON = await getCollectionJSON()
+    const accessToken = currentUser.value?.accessToken
 
-  reader.onload = ({ target }) => {
-    const content = target!.result as string | null
-    if (!content) {
-      hasFile.value = false
-      toast.show(t("action.choose_file").toString())
+    if (!accessToken) {
+      toast.error(t("error.something_went_wrong"))
+      isHoppGistCollectionExporterInProgress.value = false
       return
     }
 
-    stepResults.value.push(content)
-    hasFile.value = !!content?.length
-  }
+    if (E.isRight(collectionJSON))
+      collectionsGistExporter(collectionJSON.right, accessToken)
 
-  reader.readAsText(inputFileToImport.files[0])
+    isHoppGistCollectionExporterInProgress.value = false
+  },
 }
 
-const fileImported = () => {
-  toast.success(t("state.file_imported").toString())
-  hideModal()
-}
-const failedImport = () => {
-  toast.error(t("import.failed").toString())
-}
-const hideModal = () => {
-  resetImport()
-  emit("hide-modal")
-}
+const importerModules = computed(() => {
+  const enabledImporters = [
+    HoppRESTImporter,
+    HoppMyCollectionImporter,
+    HoppOpenAPIImporter,
+    HoppPostmanImporter,
+    HoppInsomniaImporter,
+    HoppGistImporter,
+  ]
+
+  const isTeams = props.collectionsType.type === "team-collections"
+
+  return enabledImporters.filter((importer) => {
+    return isTeams
+      ? importer.metadata.applicableTo.includes("team-workspace")
+      : importer.metadata.applicableTo.includes("personal-workspace")
+  })
+})
+
+const exporterModules = computed(() => {
+  const enabledExporters = [
+    HoppMyCollectionsExporter,
+    HoppTeamCollectionsExporter,
+  ]
+
+  if (platform.platformFeatureFlags.exportAsGIST) {
+    enabledExporters.push(HoppGistCollectionsExporter)
+  }
+
+  return enabledExporters.filter((exporter) => {
+    return exporter.metadata.applicableTo.includes(
+      props.collectionsType.type == "my-collections"
+        ? "personal-workspace"
+        : "team-workspace"
+    )
+  })
+})
+
+const hasTeamWriteAccess = computed(() => {
+  const { collectionsType } = props
 
-const resetImport = () => {
-  importerType.value = null
-  hasFile.value = false
-  hasGist.value = false
-  stepResults.value = []
-  inputChooseFileToImportFrom.value = ""
-  inputChooseGistToImportFrom.value = ""
-  mySelectedCollectionID.value = undefined
+  const isTeamCollection = collectionsType.type === "team-collections"
+
+  return isTeamCollection
+    ? collectionsType.selectedTeam?.myRole == "EDITOR" ||
+        collectionsType.selectedTeam?.myRole == "OWNER"
+    : false
```

Comment:
I think in this case, since there is a condition chain, `if` would be appropriate. Return early if `isTeamCollection` is `false`; else, return the value yield from the role check. If you prefer to keep the ternary version, maybe using `Array.includes` for the role check might suit here.